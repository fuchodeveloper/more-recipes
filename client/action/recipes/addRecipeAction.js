import { batchActions } from 'redux-batched-actions';
import axios from 'axios';
import alertify from 'alertify.js';
import { ADD_RECIPE, ADD_RECIPE_ERROR } from '../types';
import { setFetching, unsetFetching } from '../fetching';

/**
 * @description create a recipe
 *
 * @export addRecipeActionCreator
 *
 * @param {Object} recipe
 *
 * @returns {Object} recipe
 */
const addRecipeActionCreator = recipe => ({
  type: ADD_RECIPE,
  recipe
});

/**
 * @description create a recipe error
 *
 * @export addRecipeActionError
 *
 * @param {Object} error
 *
 * @returns {Object} error
 */
const addRecipeActionError = error => ({
  type: ADD_RECIPE_ERROR,
  error
});

const addRecipeAction = recipe => (dispatch) => {
  dispatch(setFetching());
  return axios.post('/api/v1/recipes', recipe)
    .then((response) => {
      alertify.delay(900);
      alertify.logPosition('bottom right');
      alertify.success(response.data.message);
      dispatch(batchActions([
        dispatch(addRecipeActionCreator(response.data.recipe)),
        unsetFetching()
      ]));
    })
    .catch((error) => {
      dispatch([
        dispatch(addRecipeActionError(error.response.data.error)),
        unsetFetching()
      ]);
    });
};

export default addRecipeAction;
