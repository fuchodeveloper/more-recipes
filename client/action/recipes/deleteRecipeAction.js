import axios from 'axios';
import alertify from 'alertify.js';
import { DELETE_RECEIPE, DELETE_RECEIPE_FAIL } from '../types';

export const deleteRecipeActionError = error => ({
  type: DELETE_RECEIPE_FAIL,
  error
});

export const deleteRecipeActionCreator = recipes => ({
  type: DELETE_RECEIPE,
  recipes
});

const deleteRecipeAction = recipeId => dispatch =>
  axios.delete(`/api/v1/recipes/${recipeId}`)
    .then((response) => {
      alertify.delay(900);
      alertify.logPosition('bottom right');
      alertify.success(response.data.message);
      dispatch(deleteRecipeActionCreator(response.data.recipes));
    })
    .catch((error) => {
      alertify.delay(900);
      alertify.logPosition('bottom right');
      alertify.success(error.data.message);
      dispatch(deleteRecipeActionError(error.data.message));
    });

export default deleteRecipeAction;
