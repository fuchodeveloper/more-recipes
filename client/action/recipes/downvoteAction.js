import axios from 'axios';
import alertify from 'alertify.js';
import { GET_DOWNVOTE } from '../types';

/**
 * Recipe doenvote action creator
 * @param {recipe} recipe
 * @returns {recipe} recipe
 */
export const downvoteRecipeActionCreator = recipe => ({
  type: GET_DOWNVOTE,
  recipe
});

/**
 * Authenticated user can downvote a recipe
 *
 * @export func
 * @param {integer} param
 * @returns {object} object
 */
const downvoteRecipe = param =>
  dispatch => axios.post(`/api/v1/recipes/${param}/downvote`)
    .then((response) => {
      alertify.delay(900);
      alertify.logPosition('bottom right');
      alertify.success(response.data.message);
      return dispatch(downvoteRecipeActionCreator(response.data.recipe));
    })
    .catch((error) => {
      alertify.logPosition('bottom right');
      alertify.error(error.message);
    });


export default downvoteRecipe;
