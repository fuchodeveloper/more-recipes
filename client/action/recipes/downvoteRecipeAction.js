import axios from 'axios';
import alertify from 'alertify.js';
import { DOWNVOTE_RECIPE, DOWNVOTE_RECIPE_ERROR } from '../types';
import networkError from '../networkError';

/**
 * @description Recipe downvote action creator
 *
 * @param {recipe} recipe recipe parameter
 *
 * @returns {recipe} recipe downvoted
 */
export const downvoteRecipeActionCreator = recipe => ({
  type: DOWNVOTE_RECIPE,
  recipe
});
/**
 * @description Recipe downvote error
 *
 * @param {Object} error error parameter
 *
 * @returns {Object} error returns the error object
 *
 */
const downvoteRecipeActionError = error => ({
  type: DOWNVOTE_RECIPE_ERROR,
  error
});

/**
 * @description Authenticated user can downvote a recipe
 *
 * @export downvoteRecipe
 *
 * @param {Number} id
 *
 * @returns {object} downvoted recipe object
 */
const downvoteRecipeAction = id =>
  dispatch => axios.post(`/api/v1/recipes/${id}/downvote`)
    .then((response) => {
      alertify.delay(900);
      alertify.logPosition('bottom right');
      alertify.success(response.data.recipe.message);
      return dispatch(downvoteRecipeActionCreator(response.data.recipe));
    })
    .catch((error) => {
      if (!error.response) {
        return networkError(error);
      }
      alertify.delay(1000);
      alertify.logPosition('bottom right');
      alertify.error('Please login to downvote recipe');
      dispatch(downvoteRecipeActionError(error.response.data));
    });


export default downvoteRecipeAction;
