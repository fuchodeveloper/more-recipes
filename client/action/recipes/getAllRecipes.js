import axios from 'axios';
import { batchActions } from 'redux-batched-actions';
import { GET_ALL_RECIPES, GET_PAGE_DETAILS } from '../types';
import { setFetching, unsetFetching } from '../fetching';

export const receiveRecipes = recipes => ({
  type: GET_ALL_RECIPES,
  recipes
});

export const recipePageDetails = pageCount => ({
  type: GET_PAGE_DETAILS,
  pageCount
});

const getAllRecipes = page => (dispatch) => {
  dispatch(setFetching());
  axios.get(`/api/v1/recipes?page=${page}`)
    .then((response) => {
      dispatch(batchActions([
        receiveRecipes(response.data.recipes),
        recipePageDetails(response.data.pageCount),
        unsetFetching(),
      ]));
    })
    .catch((error) => {
      // dispatch(batchActions([

      // ]));
      console.log(error.message);
    });
};

export default getAllRecipes;
