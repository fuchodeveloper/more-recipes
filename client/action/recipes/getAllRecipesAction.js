import axios from 'axios';
import { batchActions } from 'redux-batched-actions';
import
{ GET_ALL_RECIPES, GET_PAGE_DETAILS, GET_ALL_RECIPES_ERROR } from '../types';
import { setFetching, unsetFetching } from '../fetching';

/**
 * @description get all recipes action creator
 *
 * @param {Object} recipes recipes object parameter
 *
 * @returns {Object} recipes returns all recipes object
 */
const receiveRecipes = recipes => ({
  type: GET_ALL_RECIPES,
  recipes
});

/**
 * @description get all recipes error action
 *
 * @param {Object} error recipes error object
 *
 * @returns {Object} error returns get all recipes error object
 */
const receiveRecipesError = error => ({
  type: GET_ALL_RECIPES_ERROR,
  error
});

/**
 * @description handles page detail
 *
 * @param {Number} pageCount recipe page count parameter
 *
 * @returns {Number} returns recipes page Count
 */
const recipePageCount = pageCount => ({
  type: GET_PAGE_DETAILS,
  pageCount
});

const getAllRecipesAction = page => (dispatch) => {
  dispatch(setFetching());
  return axios.get(`/api/v1/recipes?page=${page}`)
    .then((response) => {
      dispatch(batchActions([
        dispatch(receiveRecipes(response.data.recipes)),
        recipePageCount(response.data.pageCount),
        unsetFetching(),
      ]));
    })
    .catch((serverError) => {
      dispatch(receiveRecipesError(serverError.recipes.data.error));
    });
};

export default getAllRecipesAction;
