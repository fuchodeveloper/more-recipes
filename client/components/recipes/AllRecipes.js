import React from 'react';
import { Link } from 'react-router-dom';
import noodles from '../../assets/img/noodles.jpg';

class AllRecipes extends React.Component {
  
  render() {
    const { details } = this.props;

    return (      
      <div className="col-md-4 mb-2">
          <div className="card mt-1">
            <img className="card-img-top" src={noodles} alt="Recipe image"/>
            <div className="card-body">
                <h4 className="card-title">{details.recipeName}</h4>
                <p className="card-text">{details.recipeDirection}</p>
                <div className="card-footer custom-card-footer-bg">
                    <p className="card-text"><small className="text-muted">{details.views} <i className="fa fa-eye" aria-hidden="true"/> . {details.upVotes} <i className="fa fa-thumbs-up" aria-hidden="true"/> . 42 <i className="fa fa-star" aria-hidden="true"/></small></p>
                    <Link to={`/api/v1/recipes/${details.id}`} className="btn btn-primary btn-primary-color">View Recipe</Link>
                </div>
            </div>
        </div>
      </div>

        

      
      
    );
  }
}

export default AllRecipes;

