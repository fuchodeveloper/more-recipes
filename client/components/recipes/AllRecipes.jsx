/* eslint-disable max-len */
import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export const AllRecipes = props => (
  <div className="col-md-4 mb-2">
    <div className="card mt-1">
      <img
        id="image"
        className="card-img-top"
        max-width="348px"
        height="231px"
        src={props.details.image === ''
        ? 'https://res.cloudinary.com/fuchodeveloper/image/upload/v1516760699/noodles_c6ltkq.jpg'
        : props.details.image}
        alt={props.details.name}
      />
      <div className="card-body">
        <h4
          className="card-title"
          id="name"
        >{props.details.name.slice(0, 20)}...
        </h4>
        <p
          className="card-text"
          id="direction"
        >
          {`${props.details.direction.slice(0, 100)}...`}
        </p>
        <div className="card-footer custom-card-footer-bg">
          <p className="card-text">
            <small className="text-muted">
              {props.details.views} &nbsp;
              <i className="fa fa-eye" aria-hidden="true" /> .
              &nbsp;{props.details.upVotes}&nbsp;
              <i className="fa fa-thumbs-up" aria-hidden="true" /> .
              &nbsp;{props.details.favoriteCount} &nbsp;
              <i className="fa fa-star" aria-hidden="true" />
            </small>
          </p>
          <Link
            to={`/recipes/${props.details.id}`}
            className="btn btn-primary btn-primary-color"
            id="view-recipe-button"
          >View Recipe
          </Link>
        </div>
      </div>
    </div>
  </div>
);

AllRecipes.defaultProps = {
  details: {
    favoriteCount: 0,
    upVotes: 0
  }
};

AllRecipes.propTypes = {
  details: PropTypes.shape({
    image: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    direction: PropTypes.string.isRequired,
    views: PropTypes.number.isRequired,
    id: PropTypes.number.isRequired,
    upVotes: PropTypes.number,
    favoriteCount: PropTypes.number,
  }),
};

export default AllRecipes;
