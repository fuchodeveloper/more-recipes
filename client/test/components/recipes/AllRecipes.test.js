/* eslint-disable max-len */
import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { AllRecipes } from '../../../components/recipes/AllRecipes';

let props;

const setup = () => {
  props = {
    details: {
      image: 'https://res.cloudinary.com/fuchodeveloper/image/upload/v1516760699/noodles_c6ltkq.jpg',
      name: 'jollof',
      direction: 'rice, beans',
      views: 0,
      id: 1
    }
  };
  return shallow(<AllRecipes {...props} />);
};

describe('AllRecipes component', () => {
  describe('AllRecipes component snapshot', () => {
    it('should render with right amount of elements', () => {
      expect(setup()).toMatchSnapshot();
    });
  });
});

