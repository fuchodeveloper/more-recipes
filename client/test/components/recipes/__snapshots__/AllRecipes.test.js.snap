// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllRecipes component AllRecipes component snapshot should render with right amount of elements 1`] = `
<div
  className="col-md-4 mb-2"
>
  <div
    className="card mt-1"
  >
    <img
      alt="jollof"
      className="card-img-top"
      height="231px"
      max-width="348px"
      src="https://res.cloudinary.com/fuchodeveloper/image/upload/v1516760699/noodles_c6ltkq.jpg"
    />
    <div
      className="card-body"
    >
      <h4
        className="card-title"
      >
        jollof
        ...
      </h4>
      <p
        className="card-text"
      >
        rice, beans...
      </p>
      <div
        className="card-footer custom-card-footer-bg"
      >
        <p
          className="card-text"
        >
          <small
            className="text-muted"
          >
            0
              
            <i
              aria-hidden="true"
              className="fa fa-eye"
            />
             .  
             
            <i
              aria-hidden="true"
              className="fa fa-thumbs-up"
            />
             .  
              
            <i
              aria-hidden="true"
              className="fa fa-star"
            />
          </small>
        </p>
        <Link
          className="btn btn-primary btn-primary-color"
          id="view-recipe-button"
          replace={false}
          to="/recipes/1"
        >
          View Recipe
        </Link>
      </div>
    </div>
  </div>
</div>
`;
